shared/actions/chat2/index.js

```diff
@@ -1277,7 +1277,7 @@ const messageSend = (action: Chat2Gen.MessageSendPayload, state: TypedState) =>
 
   logger.info('[MessageSend]', 'non-empty text?', text.stringValue().length > 0)
 
-  return Saga.sequentially([addMessage, postText])
+  return Saga.sequentially([addMessage, postText, addMessage])
```

Comment:
The reason we do this twice is to "undo" the removal of the first `addMessage` by a callback from `GetThreadNonblock` that was live while we dispatched these actions. The change to `upgradeMessage` complements this by making it safe to send in the message with the same `outboxID` as one that may be in the store as a sent message already.

Reply:
i this this is very confusing, so at least we have to add a comment why we'd do this but really this is cuase we don't have the bookkeeping to fix this race so we're injecting it in twice, which seems like a way to paper over the symptom

Reply:
I agree it's not ideal, but given the options seems like the easiest way to handle the situation. Added a comment.